modules/mcc/include/opencv2/mcc/colorspace.hpp

```diff
@@ -0,0 +1,635 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+//
+//                       License Agreement
+//              For Open Source Computer Vision Library
+//
+// Copyright(C) 2020, Huawei Technologies Co.,Ltd. All rights reserved.
+// Third party copyrights are property of their respective owners.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//             http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Longbu Wang <wanglongbu@huawei.com.com>
+//         Jinheng Zhang <zhangjinheng1@huawei.com>
+//         Chenqi Shan <shanchenqi@huawei.com>
+
+#ifndef __OPENCV_MCC_COLORSPACE_HPP__
+#define __OPENCV_MCC_COLORSPACE_HPP__
+
+#include <vector>
+#include <string>
+#include <iostream>
+#include "opencv2/mcc/io.hpp"
+#include "opencv2/mcc/operations.hpp"
+#include "opencv2/mcc/utils.hpp"
+
+namespace cv
+{
+namespace ccm
+{
+/* *\ brief Basic class for ColorSpace.
+*/
+class ColorSpace
```

Comment:
all the exported classes should use `CV_EXPORTS` macro; look at the other exported from OpenCV classes